src/rgw/rgw_sal_motr.cc

```diff
@@ -2766,9 +2790,19 @@ int MotrMultipartUpload::list_parts(const DoutPrefixProvider *dpp, CephContext *
   vector<string> key_vec(num_parts);
   vector<bufferlist> val_vec(num_parts);
 
-  std::string oid = mp_obj.get_key();
   string tenant_bkt_name = get_bucket_name(bucket->get_tenant(), bucket->get_name());
-  string obj_part_iname = "motr.rgw.object." + tenant_bkt_name + "." + oid + ".parts";
+  string upload_id = get_upload_id();
+
+  if(upload_id.length() == 0){
+    rc = store->get_upload_id(tenant_bkt_name, this->get_key(), upload_id);
```

Comment:
why we are calling get_upload_id() twice ?

Reply:
@sachinpunadikar string upload_id = get_upload_id(); here get_upload_id() is a member function of class MotrMultipartUpload, this upload id is correct when MultipartUpload is active for other cases this upload id is an empty string.
store->get_upload_id(tenant_bkt_name, this->get_key(), upload_id); This is a member function of MotrStore class, this is called in case upload id retrieved from MotrMultipartUpload class is empty string which indicates that upload has completed. This function can be called from other classes as well without the need for MotrMultipartUpload class.